### ISSUE TITLE
Ditch google notifications

### ISSUE THREAD

**Issue by ghost** [ProtonMail/WebClients#359]
Hi,

It seems weird that PM still relies on GCM/FCM for notifications.

Even with PM APK downloaded from the website, and having whitelisted PM in DNSfilter ([personalDNSfilter | F-Droid - Free and Open Source Android App Repository 3](https://f-droid.org/en/packages/dnsfilter.android/)), I can’t receive PM notifications if I do not open the app, because I blocked GCM/FCM adresses ([mtalk.google.com 5](http://mtalk.google.com/), [alt1-mtalk.google.com](http://alt1-mtalk.google.com/),…) with DNSfilter.

While we trust PM for keeping the minimum metadata, how can we know what Google keeps if PM needs to transmit data to their servers for each notification/message?

Furthermore, curcumventing GCM/FCM has been solved by other android app team, like Tutanota (the blog post also cites Mastodon):

[Link to post in blog](https://f-droid.org/en/2018/09/03/replacing-gcm-in-tutanota.html)

Is there a plan to drop GCM/FCM (which as Tutanota highlights “includes Google’s tracking code for analytics purposes”) for SSE (Server Sent Events API) or another alternative ?

**Comment by o0nd7ots** [comment]
Also requiring Google Play Services for actual notifications makes the app less usable on a degoogled Android rom.

**Comment by RokeJulianLockhart** [comment]
@o0nd7ots, I agree. https://ntfy.sh/ is what we should replace it with. It's the FOSS standard, at least on F-Droid. See https://protonmail.uservoice.com/forums/945460-general-ideas/suggestions/19635550-remove-dependance-on-google-play-services.

**Comment by geekley** [comment]
UnifiedPush works on de-googled devices and can even fallback to Google/FCM on other devices.
And it seems the ntfy server mentioned above can also be self-hosted.
For more info, see:
https://unifiedpush.org/
https://f-droid.org/2022/12/18/unifiedpush.html

Another option is to look at how FairEmail does it. I don't think it uses UnifiedPush, but it works great on a phone with google services disabled. I think it just keeps running a foreground service?

**Comment by UlyssesZh** [comment]
FairEmail and K-9/Thunderbird run a foreground service to keep alive to receive notification, which is less decent than UnifiedPush because of the battery consumption. To them, UnifiedPush isn't an option because they do not control the mail servers. ProtonMail client can support UnifiedPush because the developers can cooperate with the developers of the backend server.

**Comment by RokeJulianLockhart** [comment]
@UlyssesZh, the unnecessarily reductive resolution of [`github.com/thunderbird/thunderbird-android/issues/5165`](https://github.com/thunderbird/thunderbird-android/issues/5165#issue-812968237) should not be considered to be a sensible example – Ntfy's UnifiedPush implementation is self-hostable and FOSS. As you state, ProtonMail already operates a significant superset of IMAP, and doesn't intend to implement JMAP soon, so UP is the most sensible implementation, whether PM operates their own server (by default) or not.
